src/Renci.SshNet/ISftpClient.cs

```diff
@@ -698,21 +698,23 @@ public interface ISftpClient : IBaseClient, IDisposable
         /// <exception cref="ObjectDisposedException">The method was called after the client was disposed.</exception>
         IEnumerable<ISftpFile> ListDirectory(string path, Action<int> listCallback = null);
 
+#if FEATURE_ASYNC_ENUMERABLE
         /// <summary>
-        /// Asynchronously retrieves list of files in remote directory.
+        /// Asynchronously enumerates the files in remote directory.
         /// </summary>
         /// <param name="path">The path.</param>
         /// <param name="cancellationToken">The <see cref="CancellationToken"/> to observe.</param>
         /// <returns>
-        /// A <see cref="Task{IEnumerable}"/> that represents the asynchronous list operation.
-        /// The task result contains an enumerable collection of <see cref="SftpFile"/> for the files in the directory specified by <paramref name="path" />.
+        /// An <see cref="IAsyncEnumerable{SftpFile}"/> that represents the asynchronous enumeration operation.
```

Comment:
As far as I'm aware the **IAsyncEnumerable{SftpFile}** syntax doesn't work. Here **SftpFile** (which should have been **ISftpFile** instead) will be considered as the name of the type argument, not a concrete type argument. You must write something in line of:
```c#
An <see cref="IAsyncEnumerable{T}"/> of <see cref="ISftpFile"/> ...
```